source/simulator/parameters.cc

```diff
@@ -302,6 +302,42 @@ namespace aspect
                        "the composition system gets solved. See 'linear solver "
                        "tolerance' for more details.");
 
+
+    prm.enter_subsection("Formulation");
+    {
+      prm.declare_entry ("Formulation", "custom",
+                         Patterns::Selection ("isothermal compression|custom|ALA|BA"),
+                         "Select a formulation for the basic equations. Different "
+                         "published formulations are available in ASPECT (see list of "
+                         "possible parameters for available options). Two ASPECT specific "
+                         "options are: 'isothermal compression': ASPECT's original formulation, "
+                         "using the incompressible or explicit compressible mass equation, "
+                         "and the full density for the temperature equation. 'custom': "
+                         "A custom selection of 'Mass density approximation' and "
+                         "'Temperature density approximation'. Warning: The 'custom' option is "
+                         "implemented for advanced users that want full control over the "
+                         "equations solved. It is possible to choose inconsistent formulations "
+                         "and no error checking is performed on the consistency of the resulting "
+                         "equations.");
```

Comment:
No, the full equations (1)-(3) would be the 'isothermal compression' option. However so far our default was to either use the `isothermal compression` option or the incompressible version depending on the material model, and when writing this PR we decided that this behavior is somewhat dangerous (because we can not check if the user actually intends to run a compressible or incompressible model). Therefore we would like to require this setting in the parameter file in the future. For now though we kept the previous default behavior, which is equivalent to 'custom' with the associated warning.